### DISCUSSION TITLE
Use amap instead of OSM in China

### DISCUSSION THREAD

**Discussion by hipudding** [teslamate-org/teslamate#3960] [Ideas]
OpenStreetMap (OSM) maps are disabled in mainland China, so the TeslaMate homepage, Grafana map plugin, and location reverse geocoding cannot function properly. I have two ideas to solve this problem:

1. For users in mainland China, reverse geocoding can use Amap's service, which provides more accurate geolocation data for mainland China. see [teslamate_fix_addrs](https://github.com/hipudding/teslamate_fix_addrs).
2. Modify the map layer of the Grafana map widget to an XYZ Tile layer (using Amap's map tiles). see [this demo](https://github.com/hipudding/teslamate_fix_addrs/issues/7)

One important issue is that most map services in mainland China use the GCJ-02 coordinate system, so GPS coordinates need to be converted. This can be addressed by using PostgreSQL stored procedures.

I would like to know the community's thoughts on these two ideas and whether such a PR would be welcomed.

**Comment by hipudding** [comment]
OSM地图在国内体验不好，最近研究了一下，Grafana中的OSM地图可以更换为高德地图，大概的思路是：

把grafana地图插件的地图图层更换为高德瓦片。
在数据库中加入存储过程，将wgs84转gcj02坐标。
先展示下成果吧：
![image](https://github.com/teslamate-org/teslamate/assets/3837401/fab80fad-88f1-4524-a494-e34cdf8ff780)

使用方法请参考[这里](https://github.com/hipudding/teslamate_fix_addrs/issues/7)

**Reply by blue-eric** [reply]
你好，这个可以出一个更详细一点的教程吗？对小白不太好理解，谢谢了

**Reply by hipudding** [reply]
参考官方文档[docker install](https://docs.teslamate.org/docs/installation/docker)，然后下载amap_patch.zip，然后执行上面的命令就行了
有时间了准备搞一个grafana的镜像，直接启动就能用的那种。或者如果社区同意，直接合入社区是最好

**Reply by blue-eric** [reply]
我的teslamate已经安装好正常运行，但是在这边导入数据时，不知道填哪个路径，尝试后失败了

**Reply by hipudding** [reply]
不需要什么路径信息啊，需要的路径都在上面的命令里了，错误信息贴一下

**Reply by blue-eric** [reply]
输入docker cp dashboards$(docker compose ps -q grafana):/命令后，返回以下信息
root@admin:/volume1/docker/teslamate/amap_patch# docker cp dashboards$(docker-compose ps -q grafana):/
"docker cp" requires exactly 2 arguments.
See 'docker cp --help'.

Usage:  docker cp [OPTIONS] CONTAINER:SRC_PATH DEST_PATH|-
	docker cp [OPTIONS] SRC_PATH|- CONTAINER:DEST_PATH

Copy files/folders between a container and the local filesystem
这个请看一下，是不是要安装指南中的/var/lib/grafana这个路径？

**Reply by hipudding** [reply]
$前少了一个空格🌚，再试试

**Reply by blue-eric** [reply]
发现命令上差个空格，现在是命令完成了，但是查看仪表板，没有更换成对应的仪表

**Reply by hipudding** [reply]
重启grafana了么？重启后刷新下浏览器试试呢

**Reply by blue-eric** [reply]
感谢，已经可以实现了
别的涉及地图的，是不是参照这个仪表更改就可以了？

**Reply by blue-eric** [reply]
就是别的页面涉及地图的，我直接修改仪表的json文件就可以了，是吗？

**Reply by hipudding** [reply]
客气，不同的地图坐标编码不同，比如说百度，如果想换百度地图的话，可能需要新增百度坐标点的转换存储过程。

**Reply by hipudding** [reply]
> 就是别的页面涉及地图的，我直接修改仪表的json文件就可以了，是吗？

teslamate涉及地图的都修改了，你看看

**Reply by blue-eric** [reply]
好的，谢谢

**Reply by blue-eric** [reply]
好的，谢谢，我还有个疑问，就是我在grafana里批量把所有仪表删除后，过一会就重新被服务区的覆盖了，这个是为啥？

**Reply by hipudding** [reply]
这个我不了解。如果你删除了grafana的卷，那这些配置就会丢失了，单纯重启不会有问题。

**Reply by blue-eric** [reply]
我的意思就是：我每次对json修改后，都会被自动复原，变成拉取的teslamate 上的仪表，包括对内容汉化，全都变回去了

**Reply by hipudding** [reply]
这个我还没有遇到过

**Reply by blue-eric** [reply]
也真是神奇了

**Reply by blue-eric** [reply]
你好，我在国内使用了第三方的仪表，可以请您帮我适配一下高德地图吗？

**Reply by hipudding** [reply]
> 你好，我在国内使用了第三方的仪表，可以请您帮我适配一下高德地图吗？

发下仪表github地址，有时间弄一下

**Reply by blue-eric** [reply]
太感谢了，这个是项目地址https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards

**Reply by blue-eric** [reply]
你好，还有我在Visited这个画面，无法显示地图

![Clipboard Image (2024-06-23 09 23 04)](https://github.com/teslamate-org/teslamate/assets/118333287/88853c25-e7d1-4585-86ae-1cbf9ad7da0b)

**Reply by hipudding** [reply]
图好像挂了，我这里visited显示正常的啊，你试试拉下新的镜像： hipudding/grafana。现在不用应用patch了，把grafana换一下就行了

**Reply by blue-eric** [reply]
我找到原因了，是默认90天的数据，加载太慢，导致无响应了

**Reply by hipudding** [reply]
是的，这个90天的数据比较多，修改成高德地图后，数据库需要算偏移坐标，计算量很大。如果设备的性能弱一些的话，显示会比较慢

**Reply by blue-eric** [reply]
ccr.ccs.tencentyun.com/dhuar/grafana 这个仪表做了比较好的汉化，如果能把这个地图做成高德的，那就更好了。

**Reply by hipudding** [reply]
> ccr.ccs.tencentyun.com/dhuar/grafana 这个仪表做了比较好的汉化，如果能把这个地图做成高德的，那就更好了。

这个我打不开，贴下github链接吧。有时间一起看下

**Reply by blue-eric** [reply]
这个作业没有github，他是卖teslamate盒子的，然后把汉化后的上传到腾讯云，免费使用的

**Reply by hipudding** [reply]
> 这个作业没有github，他是卖teslamate盒子的，然后把汉化后的上传到腾讯云，免费使用的

没有源码就不知道怎么搞了。这个网址我也打不开。
不过拿开源软件过来卖钱，鄙视下这个人

**Reply by blue-eric** [reply]
目前没办法，只能拉取，或者我这有个源码包呢，各取所需吧，有的人不会弄，就买他个盒子，会弄的就拉取他的源码就可以了，哈哈

**Reply by blue-eric** [reply]
或者大佬，能不能用你的包，我来协助汉化，你来发包呀？我把grafana的仪表尽量的汉化，然后把json文件发你

**Reply by hipudding** [reply]
欢迎，给[这个库](https://github.com/hipudding/teslamate)提PR吧，这个库是修改过的

**Reply by blue-eric** [reply]
我不会在这个上面提，我只会死办法，一个一个修改后，导出json。。哎

**Reply by blue-eric** [reply]
你好，请问有时间看一下之前的汉化版的文件了吗？如果好了，是不是直接拉取你的项目库就可以了呢？

**Reply by hipudding** [reply]
> 你好，请问有时间看一下之前的汉化版的文件了吗？如果好了，是不是直接拉取你的项目库就可以了呢？

我不知道汉化版的文件在哪里，能贴一个链接么？

**Reply by blue-eric** [reply]
可能我表达错了，我是希望你帮我们做个汉化版，可以吗？国内用户目前使用都比较困难

**Reply by hipudding** [reply]
了解，最近比较忙，还没顾上看，有时间了一起看下

**Reply by blue-eric** [reply]
我可以添加你即时通讯吗？我想请教一些问题，也想做一些汉化工作，以便自己和于国内车友更好的使用。

**Reply by hipudding** [reply]
没问题，加我微信吧 id： huafengchun

**Comment by Dalishimin** [comment]
看到中文，刚好请教一下您。
1.我已经升级docker到V2以上，为啥还是用不了您的这种命令，我还是必须有个-。docker compose exec
> docker-compose --version
> Docker Compose version v2.9.0-6413-g38f6acd

2.现在我升级pull不了，科学后也不行。传说docker镜像站前几天被封，不知道怎么解决 [提交了一下反馈](https://github.com/teslamate-org/teslamate/discussions/3956)

**Reply by hipudding** [reply]
有的版本的docker compose是docker的一个子命令，使用方法是docker compose xxxxx，有的是一个脚本，脚本名字叫 docker-compose，功能都是一样的，有哪个用哪个就行了。

dockerhub国内应该都凉了，科学上网后是可以pull的，检查下docker配置，参考下[这里](https://gist.github.com/y0ngb1n/7e8f16af3242c7815e7ca2f0833d3ea6)，看还有没有能用的源

**Reply by Dalishimin** [reply]
谢谢，我研究一下。

**Comment by ArvinLiYongJun** [comment]
你好，我是程序小白，有没有傻瓜式的操作 XD

**Reply by hipudding** [reply]
使用方法

使用修改的grafana
将docker-compose中的teslamate/grafana 修改为 hipudding/grafana

重启

docker compose restart

**Comment by Wuchangwen** [comment]
请教一下，现在这个项目还在维护吗=-=

**Comment by JakobLichterfeld** [comment]
Only english
